wordpress-mobile/WordPress-iOS
# Fix/17344 - Close account error messages

Fixes [#17344](https://github.com/wordpress-mobile/WordPress-iOS/issues/17344)

This PR adds more specific localized error message for different error types that can be encountered while trying to close the user account.

## Test closing account with active subscription:
1. Make sure logged in user has an active subscription
2. Go to **My Site > Profile > Account Settings > Close account**
3. Enter the username when prompted
3. Closing account should fail with the following error message:

<img width="502" alt="Screenshot 2021-10-21 at 15 11 21" src="https://user-images.githubusercontent.com/12729242/138287433-2da8458e-3ae8-4430-9b4d-d558f8bc13b3.png">

## Test closing account with atomic site
1. Make sure logged in user has an atomic site, or is added as an admin to one
2. Go to **My Site > Profile > Account Settings > Close account**
3. The following error message should be displayed:

<img width="502" alt="Screenshot 2021-10-21 at 15 14 07" src="https://user-images.githubusercontent.com/12729242/138287729-452e3fef-40d0-454a-8dc7-be142ae336ce.png">

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:
